### ISSUE TITLE
Layered test checker doesn't diagnose three-layer cakes

### ISSUE THREAD

**Issue by nedbat** [edx/edx-lint#48]
This code runs two tests in nose and pytest:
```


# Three-layer cake.

class BaseTest(unittest.TestCase):
    __test__ = False
    def test_it(self):
        self.fail("Hello, world!")

class Middle(BaseTest):
    __test__ = True

class MoreTests(Middle):
    __test__ = True
```
but the layered test checker doesn't detect it.
